yasnippet.el

```diff
@@ -1352,7 +1352,7 @@ Returns (TEMPLATES START END). This function respects
       ((debug error) (cdr oops)))))
 
 (defun yas--eval-for-effect (form)
-  (yas--safely-run-hook (apply-partially #'eval form)))
```

Comment:
We might need to bind `debug-on-error` based `yas-good-grace` similar to what `yas--safely-run-hook` does.